src/main/kotlin/net/weavemc/loader/WeaveLoader.kt

```diff
@@ -55,25 +56,30 @@ public object WeaveLoader {
 
                 val configEntry = jar.getEntry("weave.mod.json") ?: error("${jar.name} does not contain a weave.mod.json!")
                 val config = json.decodeFromStream<ModConfig>(jar.getInputStream(configEntry))
+                val name = config.name ?: jar.name
 
                 config.mixinConfigs.forEach(Mixins::addConfiguration)
                 HookManager.hooks += config.hooks.map(::instantiate)
-                initializers += config.entrypoints.map(::instantiate)
-            }
 
-        // call preInit after all hooks/mixins are added
-        initializers.forEach { it.preInit() }
+                // TODO: Add a name field to the config.
+                mods[config.modId] = WeaveMod(config.entrypoints.map(::instantiate), name, config)
+            }
 
-        mods = initializers.toList()
+        // Call preInit() once everything is done.
+        mods.values.forEach {
+            it.instance.forEach(ModInitializer::preInit)
+        }
 
         println("[Weave] Initialized Weave")
     }
 
     @Serializable
-    private data class ModConfig(
+    public data class ModConfig(
         val mixinConfigs: List<String> = listOf(),
         val hooks: List<String> = listOf(),
-        val entrypoints: List<String>
+        val entrypoints: List<String>,
+        val name: String? = null,
```

Comment:
Why is there a need for a name field in the config?

Reply:
For dependencies managing later?